discovery-provider/src/tasks/update_track_is_available.py

```diff
@@ -0,0 +1,250 @@
+        spID_unavailable_tracks_key = get_unavailable_tracks_redis_key(node["spID"])
+
+        # Clear redis for existing data
+        redis.delete(spID_unavailable_tracks_key)
+
+        for i in range(0, len(unavailable_track_ids), BATCH_SIZE):
+            unavailable_track_ids_batch = unavailable_track_ids[i : i + BATCH_SIZE]
+            redis.sadd(spID_unavailable_tracks_key, *unavailable_track_ids_batch)
+
+            # Aggregate a set of unavailable tracks
+            redis.sadd(ALL_UNAVAILABLE_TRACKS_REDIS_KEY, *unavailable_track_ids_batch)
+
+
+def update_tracks_is_available_status(db: Any, redis: Any) -> None:
+    """Check track availability on all unavailable tracks and update in Tracks table"""
+    all_unavailable_track_ids = _get_redis_set_members_as_list(
+        redis, ALL_UNAVAILABLE_TRACKS_REDIS_KEY
+    )
+
+    for i in range(0, len(all_unavailable_track_ids), BATCH_SIZE):
+        unavailable_track_ids_batch = all_unavailable_track_ids[i : i + BATCH_SIZE]
+        with db.scoped_session() as session:
+            try:
+
+                track_ids_to_replica_set = query_replica_set_by_track_id(
+                    session, unavailable_track_ids_batch
+                )
+
+                track_id_to_is_available_status = {}
+
+                entry: Union[Tuple[int, int, List[int]], Tuple[int, None, List[None]]]
+                for entry in track_ids_to_replica_set:
+                    track_id = entry[0]
+
+                    # Some users are do not have primary_ids or secondary_ids
+                    # If these values are null, default to track is available
+                    if entry[1] is None or entry[2][0] is None or entry[2][1] is None:
+                        is_available = True
+                    else:
+                        spID_replica_set = [entry[1], *entry[2]]
+                        is_available = check_track_is_available(
+                            redis=redis,
+                            track_id=track_id,
+                            spID_replica_set=spID_replica_set,
+                        )
+
+                    track_id_to_is_available_status[track_id] = is_available
+
+                # Update tracks with is_available status
+                tracks = query_tracks_by_track_ids(session, unavailable_track_ids_batch)
+                for track in tracks:
+                    is_available = track_id_to_is_available_status[track.track_id]
+
+                    # If track is not available, also flip 'is_delete' flag to True
+                    if not is_available:
+                        track.is_available = False
+                        track.is_delete = True
+
+                session.commit()
+            except Exception as e:
+                logger.warn(
+                    f"update_track_is_available.py | Could not process batch {unavailable_track_ids_batch}: {e}\nContinuing..."
+                )
+                session.rollback()
```

Comment:
the scoped_session takes care of commit / rollback for us so we don't have to :)
https://github.com/AudiusProject/audius-protocol/blob/cb0705a4366671e8ff195891f7782af83b0b1fc4/discovery-provider/src/utils/session_manager.py#L75